compiler/noirc_evaluator/src/ssa/ssa_gen/context.rs

```diff
@@ -332,15 +332,6 @@ impl<'a> FunctionContext<'a> {
 
         let mut result = self.builder.insert_binary(lhs, op, rhs);
 
-        // Check for integer overflow
-        if matches!(operator, |BinaryOpKind::ShiftLeft| BinaryOpKind::ShiftRight) {
-            let result_type =
-                self.builder.current_function.dfg.type_of_value(result).unwrap_numeric();
-            if let NumericType::Signed { bit_size } = result_type {
-                result = self.builder.insert_truncate(result, bit_size, bit_size + 1);
```

Comment:
I do think we still want to truncate here. What would happen if we have an overflowing shift left in ACIR?

Reply:
> Where do we do this?

Ah, it's true. I guess the truncations are needed then, even though we are operating with non-fields, because in the end they are all fields? It's a bit unintuitive but maybe it's done for performance reasons. For now I'll abandon this PR and the other one (I was trying to simplify the doc comments I was writing if we were able to simplify the code).

Thanks!